src/DocTest/Parser.elm

```diff
@@ -11,15 +11,31 @@ parse str =
     let
         ( imports, tests ) =
             parseComments str
-                |> List.concatMap (filterNotDocTest << parseDocTests << String.lines << .match)
+                |> List.concatMap
+                    (.match
+                        >> String.lines
+                        >> List.concatMap splitOneLiners
+                        >> parseDocTests
+                    )
                 |> List.partition isImport
     in
-        { imports = List.map toStr imports
-        , tests =
-            tests
-                |> List.Extra.groupWhile (\x y -> not <| isAssertion y)
-                |> List.filterMap toTest
-        }
+    { imports = List.map toStr imports
+    , tests =
+        tests
+            |> List.foldr collapseAssertions []
+            |> List.Extra.groupWhile (\x y -> not <| isAssertion y)
+            |> List.map filterNotDocTest
+            |> List.filterMap toTest
+    }
+
+
+splitOneLiners : String -> List String
+splitOneLiners str =
+    str
+        |> Regex.replace Regex.All
+            (Regex.regex "\\s\\-\\->\\s")
+            (\_ -> "\n    --> ")
```

Comment:
this function seems strange enough that a name would really help it make sense, IMO